src/app/index.ts

```diff
@@ -160,37 +196,40 @@ module.exports = yo.extend({
         default: 'Excel',
         choices: jsonData.getHostTemplateNames().map(host => ({ name: host, value: host })),
         when: (this.options.host == null || this.options.host != null && !jsonData.isValidInput(this.options.host, true /* isHostParam */))
-        && !isExcelFunctionsProject
+          && !isExcelFunctionsProject
       }];
       let answerForHost = await this.prompt(askForHost);
       let endForHost = (new Date()).getTime();
       let durationForHost = (endForHost - startForHost) / 1000;
 
+      usageDataObject = new usageData.OfficeAddinUsageData(usageDataOptions);
+
       /* Configure project properties based on user input or answers to prompts */
       this._configureProject(answerForProjectType, answerForScriptType, answerForHost, answerForName, isManifestProject, isExcelFunctionsProject);
-
-      /* Gnerate Insights logging */
-      const noElapsedTime = 0;
-      insight.trackEvent('Name', { Name: this.project.name }, { durationForName });
-      insight.trackEvent('Host', { Host: this.project.host }, { durationForHost });
-      insight.trackEvent('ScriptType', { ScriptType: this.project.scriptType }, { noElapsedTime });
-      insight.trackEvent('IsManifestOnly', { IsManifestOnly: this.project.isManifestOnly.toString() }, { noElapsedTime });
-      insight.trackEvent('ProjectType', { ProjectType: this.project.projectType }, { durationForProjectType });
+      const projectInfo = {
+        Host: [this.project.host, durationForHost],
+        ScriptType: [this.project.scriptType],
+        IsManifestOnly: [this.project.isManifestOnly.toString()],
+        ProjectType: [this.project.projectType, durationForProjectType],
+        IsProdData: [isProdData]
+      };
+      // Send usage data for project created
+      usageDataObject.reportEvent(defaults.promptSelectionstEventName, projectInfo);
     } catch (err) {
-      insight.trackException(new Error('Prompting Error: ' + err));
+      usageDataObject.reportError(isProdData ? defaults.promptSelectionsErrorEventName : defaults.testPromptSelectionsErrorEventName, new Error('Prompting Error: ' + err));
```

Comment:
You might consider having reportError() be able to append "-test" to the event name when isForTesting is true. You wouldn't have to make different event names for every event and write code to test it on the call. Rather the function implementation can know and do it automatically.

Reply:
great idea